### PR TITLE
Ensure no continued intake 180019127

### PR DESCRIPTION
**Pull request by bytheway875**
Screen Shot 2021-11-02 at 5 45 04 PM

This PR also prevents using "go back" and browser back to make changes to a completed submission.

### CODE REVIEW

**Comment by tgrathwell** [comment]
`redirect_if_completed_intake_present` is sorely needed and will be a great boon to the people of the earth

not sure what the part about deleting intake from the session is... did it just seem like something that ought to be cleaned up, or does it solve a particular problem?

File: app/controllers/concerns/ctc/after_verification_concern.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ module AfterVerificationConcern
     def after_verification_actions
       send_mixpanel_event(event_name: "ctc_contact_verified")
       sign_in current_intake.client
+      session.delete("intake_id")
```
Comment (alaroia-cfa) [inline app/controllers/concerns/ctc/after_verification_concern.rb:10]: I like this, for clarity.

File: app/controllers/clients/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Clients::SessionsController < Devise::SessionsController
+  def destroy
+    session[:intake_id] = nil
```
Comment (alaroia-cfa) [inline app/controllers/clients/sessions_controller.rb:3]: I'd like to know, on logout -- do we clear the entire session? If so, this isn't needed. Maybe we can Tuple and watch the network console? Harmless to keep it.

**Review by alaroia-cfa — APPROVED** [review]
One question; no blockers; yay.

File: app/controllers/clients/sessions_controller.rb (hunk 2 above)
Comment (bytheway875) [reply app/controllers/clients/sessions_controller.rb:3]: yeah, i actually think maybe this is not needed when we remove the intake_id when someone is logged in. I can also remove this code and leave the test and see if it clears.

**Comment by bytheway875** [comment]
@alaroia-cfa I tested to ensure that the entire session cookie gets deleted which it does (which def makes sense), so the custom destroy action was redundant like you said. I removed it, but I left the deletion of intake_id from the session when someone logs in because i think it makes it more straightforward that you need to look at the current_client's intake in cases where current_client is present.

**Pull request merged**